pyqtgraph/functions_qimage.py

```diff
@@ -40,6 +40,32 @@ def _apply_lut_for_uint16_mono(xp, image, lut):
     return image, augmented_alpha
 
 
+def _convert_lut_to_rgba(xp, lut):
+    # converts:
+    #   - None to (256, 4)
+    #   - uint8 (N,) to uint8 (N, 4)
+    #   - uint8 (N, 1) to uint8 (N, 4)
+    #   - uint8 (N, 3) to uint8 (N, 4)
+
+    N = lut.shape[0] if lut is not None else 256
+
+    if lut is None:
+        lut = xp.arange(N, dtype=xp.uint8)
+
+    # convert (N,) to (N, 1)
+    if lut.ndim == 1:
+        lut = lut[:, xp.newaxis]
+
+    assert lut.ndim == 2 and lut.shape[1] in (1, 3, 4)
```

Comment:
Is an `assert` statement what we actually want here? These are bypassed when Python is run with the `-O` argument.

Reply:
I am not so sure that this makes more sense as an exception. Sometimes I use assertions as comments to document the code flow.

I converted it to an exception but I moved it to the top of the function, otherwise it doesn't make sense to raise a complaint that the lut was not 2d.